gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -393,11 +397,23 @@ public void remove(Spec spec, Properties headers) throws IOException {
     // .. this will work for Identity compiler but not always for multi-hop.
     // Note: Current logic assumes compilation is consistent between all executions
     if (spec instanceof FlowSpec) {
+      URI specUri = spec.getUri();
       //Send the dag to the DagManager to stop it.
       //Also send it to the SpecProducer to do any cleanup tasks on SpecExecutor.
       if (this.dagManager.isPresent()) {
-        _log.info("Forwarding cancel request for flow URI {} to DagManager.", spec.getUri());
-        this.dagManager.get().stopDag(spec.getUri());
+        _log.info("Forwarding cancel request for flow URI {} to DagManager.", specUri);
+        this.dagManager.get().stopDag(specUri);
+      }
+      if (this.dagProcessingEngine.isPresent()) {
```

Comment:
I don't think we should be interacting with dagProcessingEngine directly, we want to use the `DagManagement` `(NewDagManager.handleCancel/Kill...)` interface to abstract this addition of a task away.